patches/0005-LPS-112982-Add-a-languageId-parameter-to-resource-UR.patch

```diff
@@ -0,0 +1,45 @@
+From 5b126a54b6e53f23345ebbcfc9741c5a48a25e31 Mon Sep 17 00:00:00 2001
+From: Julien Castelain <julien.castelain@liferay.com>
+Date: Mon, 6 Jul 2020 09:39:27 +0200
+Subject: [PATCH 5/5] LPS-112982 Add a "languageId" parameter to resource URLs
+
+---
+ core/ckeditor_base.js | 20 +++++++++++++-------
+ 1 file changed, 13 insertions(+), 7 deletions(-)
+
+diff --git a/core/ckeditor_base.js b/core/ckeditor_base.js
+index 3fb18f5f7..8e6f33018 100644
+--- a/core/ckeditor_base.js
++++ b/core/ckeditor_base.js
+@@ -171,15 +171,21 @@ if ( !window.CKEDITOR ) {
+ 			 * @returns {String} The full URL.
+ 			 */
+ 			getUrl: function( resource ) {
+-				// If this is not a full or absolute path.
+-				if ( resource.indexOf( ':/' ) == -1 && resource.indexOf( '/' ) !== 0 )
+-					resource = this.basePath + resource;
++				var resourceUrl = resource.includes(CKEDITOR.basePath)
++						? new URL(resource)
++						: new URL(CKEDITOR.basePath + resource);
+ 
+-				// Add the timestamp, except for directories.
+-				if ( this.timestamp && resource.charAt( resource.length - 1 ) != '/' && !( /[&?]t=/ ).test( resource ) )
+-					resource += ( resource.indexOf( '?' ) >= 0 ? '&' : '?' ) + 't=' + this.timestamp;
++				if (!resourceUrl.toString().endsWith('/')) {
++					if (!resourceUrl.searchParams.has('t')) {
++						resourceUrl.searchParams.set('t', CKEDITOR.timestamp);
++					}
++
++					if (window.themeDisplay && !resourceUrl.searchParams.has('languageId')) {
++						resourceUrl.searchParams.set('languageId', window.themeDisplay.getLanguageId());
++					}
++				}
```

Comment:
It looks like you dropped the `this.timestamp` check; I presume that was there for a reason though and should probably be added back, right?

Other than that, looks fine to me.

Reply:
Somehow GitHub ate my other comment here 😢 — in any case, it was that you'd changed from `this.timestamp` to `CKEDITOR.timestamp`; is that the same thing?

Reply:
Yes, in this function, `this` is `CKEDITOR`,

The parameter for `CKEDITOR.timestamp` is `t`, i.e. the URL would look something like this
`http://localhost:8000/ckeditor/styles.js?t=K669`